src/OracleService/Protocols/OracleHttpsProtocol.cs

```diff
@@ -48,17 +48,24 @@ public async Task<(OracleResponseCode, string)> ProcessAsync(Uri uri, Cancellati
         {
             Utility.Log(nameof(OracleHttpsProtocol), LogLevel.Debug, $"Request: {uri.AbsoluteUri}");
 
-            if (!Settings.Default.AllowPrivateHost)
-            {
-                IPHostEntry entry = await Dns.GetHostEntryAsync(uri.Host);
-                if (entry.IsInternal())
-                    return (OracleResponseCode.Forbidden, null);
-            }
-
             HttpResponseMessage message;
             try
             {
-                message = await client.GetAsync(uri, HttpCompletionOption.ResponseContentRead, cancellation);
+                do
+                {
+                    if (!Settings.Default.AllowPrivateHost)
+                    {
+                        IPHostEntry entry = await Dns.GetHostEntryAsync(uri.Host, cancellation);
```

Comment:
It seems that we should check `message.RequestMessage.RequestUri` instead of `uri`.


> RequestMessage is set to the request message which led to this response message. In the case of a request sent using HttpClient, this property will point to the actual request message leading to the final response. Note that this may not be the same message the user provided when sending the request. This is typically the case if the request needs to be resent due to redirects or authentication. This property can be used to determine what URL actually created the response (useful in case of redirects).

> Following a redirection header does not change the original RequestUri property. 

Copied from
- [RequestMessage](https://docs.microsoft.com/et-ee/dotnet/api/system.net.http.httpresponsemessage.requestmessage?view=netframework-4.7.2#system-net-http-httpresponsemessage-requestmessage)
- [RequestUri](https://docs.microsoft.com/et-ee/dotnet/api/system.net.httpwebrequest.requesturi?view=netframework-4.7.2&viewFallbackFrom=netstandard-1.3)

Reply:
the same problem as https://github.com/neo-project/neo-modules/pull/694#discussion_r817353589